doc/BUILD.rst

```diff
@@ -0,0 +1,722 @@
+toolchain.
+
+On Linux, you'll need the following tools for cross-compiling Taisei for Windows
+on Linux:
+
+- ``llvm-mingw``
+- `nsis <https://nsis.sourceforge.io/Main_Page>`__ >= 3.0
+
+On macOS, you're probably better off using Docker and the
+`Docker container <https://hub.docker.com/r/mstorsjo/llvm-mingw/>`__ that
+``llvm-mingw`` provides, and installing ``nsis`` on top of it.
+
+Another options for Windows-based computers is leveraging Windows
+10's
+`Windows For Linux (WSL) Subsystem <https://docs.microsoft.com/en-us/windows/wsl/install-win10>`__
+to cross-compile to Windows using their Ubuntu image.
+
+Checking Out Code
+-----------------
+
+First, you'll need to checkout the repository. You can do that with the
+following:
+
+.. code:: sh
+
+   git clone https://github.com/taisei-project/taisei.git
+   cd taisei/
+   git submodule update --init --recursive
+
+The ``git submodule update --init --recursive`` line is absolutely necessary,
+or Taisei will not build, as it will be missing many of the dependencies its
+needs to compile.
+
+Build Options
+-------------
+
+This is *not* an exhaustive list. You can see the full list of option using
+``meson`` in the ``taisei`` directory.
+
+.. code:: sh
+
+   cd taisei/
+   meson configure
+
+Setup
+"""""
+
+The first command you'll need to run is ``setup``, which creates a directory
+(in this case, ``taisei/build/``). It checks your system for various
+dependencies and required tools, which should take about a minute on most
+systems.
+
+.. code:: sh
+
+   # inside the taisei/ directory you cloned before
+   meson setup -C build/
+
+You can also have the ``setup`` command contain certain build options (seen
+below). The following are an *example* and *not required* for getting Taisei
+building.
+
+.. code:: sh
+
+   # enables Developer Mode and debugging symbols
+   meson setup -C build/ -Ddeveloper=true -Dbuildtype=debug
+
+You can then apply more build options later using ``meson configure`` (as seen
+below). It will automatically reconfigure your build environment with the new
+options without having to rebuild everything.
+
+System/Vendored Dependencies (``--wrap-mode``)
+""""""""""""""""""""""""""""""""""""""""""""""
+
+See: `Meson Manual <https://mesonbuild.com/Wrap-dependency-system-manual.html>`__
+
+* Default: ``default``
+* Options: ``default``, ``nofallback``, ``forcefallback``, ...
+
+This is a core ``meson`` flag that does quite a few things. Not all of them will
+be covered here. Refer to the ``meson`` documentation linked above.
+
+Generally, ``default`` will rely on system-installed libraries when available,
+and fallback to vendored in-repository dependencies when necessary.
+
+``forcefallback`` will heavily encourage the use of in-repository dependencies
+whenever possible. Recommended for release builds.
+
+``nofallback`` discourages the use of in-repository dependencies whenever
+possible, instead relying on system libraries. Useful for CI.
```

Comment:
Nitpick, but the use of "encourage"/"discourage" is a bit confusing here, as these imply a mild suggestion instead of hard enforcement, which is what these options actually do.